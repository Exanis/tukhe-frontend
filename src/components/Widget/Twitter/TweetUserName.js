import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Icon from '@material-ui/core/Icon';
import ConditionalComponent from "../../../components/ConditionalComponent/index";
import PropTypes from 'prop-types';


export const classes = theme => ({
    spaced: {
        display: 'inline-block',
        marginRight: 5
    },
    icon: {
        fontSize: 12,
        color: theme.palette.primary.light,
        marginRight: 5
    },
});

export class RawTweetUserName extends Component {
    render() {
        return <div>
            <Typography component="span" variant="subtitle1" inline className={this.props.classes.spaced}>{this.props.user.name}</Typography>
            <ConditionalComponent render={this.props.user.protected}>
                <Icon className={this.props.classes.icon}>lock</Icon>
            </ConditionalComponent>
            <ConditionalComponent render={this.props.user.verified}>
                <Icon className={this.props.classes.icon}>verified_user</Icon>
            </ConditionalComponent>
            <Typography component="span" variant="caption" inline color={'textSecondary'} className={this.props.classes.spaced}>@{this.props.user.screen_name}</Typography>
        </div>;
    }
}

const TweetUserName = withStyles(classes)(RawTweetUserName);

TweetUserName.propTypes = {
    user: PropTypes.shape({
        protected: PropTypes.bool.isRequired,
        verified: PropTypes.bool.isRequired,
        name: PropTypes.string.isRequired,
        screen_name: PropTypes.string.isRequired
    }).isRequired
};

RawTweetUserName.propTypes = {
    classes: PropTypes.object.isRequired,
    ...TweetUserName.propTypes
};

export default TweetUserName;
