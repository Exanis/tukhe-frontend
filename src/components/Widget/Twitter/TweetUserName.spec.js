import React from 'react';
import { shallow } from 'enzyme';
import { classes, RawTweetUserName } from "./TweetUserName";

const props = {
    user: {
        protected: true,
        verified: true,
        name: 'Test user',
        screen_name: 'test_user'
    },
    classes: {}
};

it('Should render correctly', () => {
    const wrapper = shallow(<RawTweetUserName {...props} />);

    expect(wrapper).toMatchSnapshot();
});

it('Should keep styling consistant', () => {
    expect(classes({
        palette: {
            primary: {
                light: '#aaa'
            }
        }
    })).toMatchSnapshot();
});
