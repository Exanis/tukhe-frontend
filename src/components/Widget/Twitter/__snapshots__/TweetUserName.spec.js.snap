// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should keep styling consistant 1`] = `
Object {
  "icon": Object {
    "color": "#aaa",
    "fontSize": 12,
    "marginRight": 5,
  },
  "spaced": Object {
    "display": "inline-block",
    "marginRight": 5,
  },
}
`;

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawTweetUserName
    classes={Object {}}
    user={
      Object {
        "name": "Test user",
        "protected": true,
        "screen_name": "test_user",
        "verified": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Typography)
          component="span"
          inline={true}
          variant="subtitle1"
        >
          Test user
        </WithStyles(Typography)>,
        <ConditionalComponent
          render={true}
        >
          <WithStyles(Icon)>
            lock
          </WithStyles(Icon)>
        </ConditionalComponent>,
        <ConditionalComponent
          render={true}
        >
          <WithStyles(Icon)>
            verified_user
          </WithStyles(Icon)>
        </ConditionalComponent>,
        <WithStyles(Typography)
          color="textSecondary"
          component="span"
          inline={true}
          variant="caption"
        >
          @
          test_user
        </WithStyles(Typography)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Test user",
          "className": undefined,
          "component": "span",
          "inline": true,
          "variant": "subtitle1",
        },
        "ref": null,
        "rendered": "Test user",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <WithStyles(Icon)>
            lock
          </WithStyles(Icon)>,
          "render": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "lock",
            "className": undefined,
          },
          "ref": null,
          "rendered": "lock",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <WithStyles(Icon)>
            verified_user
          </WithStyles(Icon)>,
          "render": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "verified_user",
            "className": undefined,
          },
          "ref": null,
          "rendered": "verified_user",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "@",
            "test_user",
          ],
          "className": undefined,
          "color": "textSecondary",
          "component": "span",
          "inline": true,
          "variant": "caption",
        },
        "ref": null,
        "rendered": Array [
          "@",
          "test_user",
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            component="span"
            inline={true}
            variant="subtitle1"
          >
            Test user
          </WithStyles(Typography)>,
          <ConditionalComponent
            render={true}
          >
            <WithStyles(Icon)>
              lock
            </WithStyles(Icon)>
          </ConditionalComponent>,
          <ConditionalComponent
            render={true}
          >
            <WithStyles(Icon)>
              verified_user
            </WithStyles(Icon)>
          </ConditionalComponent>,
          <WithStyles(Typography)
            color="textSecondary"
            component="span"
            inline={true}
            variant="caption"
          >
            @
            test_user
          </WithStyles(Typography)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test user",
            "className": undefined,
            "component": "span",
            "inline": true,
            "variant": "subtitle1",
          },
          "ref": null,
          "rendered": "Test user",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <WithStyles(Icon)>
              lock
            </WithStyles(Icon)>,
            "render": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "lock",
              "className": undefined,
            },
            "ref": null,
            "rendered": "lock",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <WithStyles(Icon)>
              verified_user
            </WithStyles(Icon)>,
            "render": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "verified_user",
              "className": undefined,
            },
            "ref": null,
            "rendered": "verified_user",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "@",
              "test_user",
            ],
            "className": undefined,
            "color": "textSecondary",
            "component": "span",
            "inline": true,
            "variant": "caption",
          },
          "ref": null,
          "rendered": Array [
            "@",
            "test_user",
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
